File: src/main/java/com/github/unldenis/hologram/line/animated/BlockALine.java
package com.github.unldenis.hologram.line.animated;

import com.github.unldenis.hologram.animation.Animation;
import com.github.unldenis.hologram.line.BlockLine;
import com.github.unldenis.hologram.line.ILine;
import java.util.Collection;
import java.util.Optional;
import org.bukkit.Location;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.bukkit.plugin.Plugin;

public class BlockALine implements ILine<ItemStack>, IAnimated {

  private final BlockLine blockLine;
  private final IAnimated animatedLine;

  public BlockALine(BlockLine blockLine, IAnimated animatedLine) {
    this.blockLine = blockLine;
    this.animatedLine = animatedLine;
  }

  public BlockLine getBlockLine() {
    return blockLine;
  }

  @Override
  public Plugin getPlugin() {
    return blockLine.getPlugin();
  }

  @Override
  public Type getType() {
    return Type.BLOCK_ANIMATED_LINE;
  }

  @Override
  public int getEntityId() {
    return blockLine.getEntityId();
  }

  @Override
  public Location getLocation() {
    return blockLine.getLocation();
  }

  @Override
  public void setLocation(Location location) {
    blockLine.setLocation(location);
  }

  @Override
  public ItemStack getObj() {
    return blockLine.getObj();
  }

  @Override
  public void setObj(ItemStack obj) {
    blockLine.setObj(obj);
  }

  @Override
  public void hide(Player player) {
    blockLine.hide(player);
  }

  @Override
  public void teleport(Player player) {
    blockLine.teleport(player);
  }

  @Override
  public void show(Player player) {
    blockLine.show(player);
  }

  @Override
  public void update(Player player) {
    blockLine.update(player);
  }

  @Override
  public void setAnimation(Animation animation, Collection<Player> seeingPlayers) {
    animatedLine.setAnimation(animation, seeingPlayers);
  }

  @Override
  public void removeAnimation() {
    animatedLine.removeAnimation();
  }

  @Override
  public Optional<Animation> getAnimation() {
    return animatedLine.getAnimation();
  }
}

File: src/main/kotlin/com/github/unldenis/hologram/line/animated/BlockALine.kt
package com.github.unldenis.hologram.line.animated

import com.github.unldenis.hologram.animation.Animation
import com.github.unldenis.hologram.line.BlockLine
import com.github.unldenis.hologram.line.ILine
import org.bukkit.Location
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack
import org.bukkit.plugin.Plugin
import java.util.*

class BlockALine(private val blockLine: BlockLine, private val animatedLine: IAnimated) : ILine<ItemStack>, IAnimated {

    override val plugin: Plugin
        get() = blockLine.plugin

    override val type: ILine.Type
        get() = ILine.Type.BLOCK_ANIMATED_LINE

    override val entityId: Int
        get() = blockLine.entityId

    override val location: Location?
        get() = blockLine.location

    override var obj: ItemStack
        get() = blockLine.obj
        set(value) {
            blockLine.obj = value
        }

    override fun setLocation(location: Location) {
        blockLine.setLocation(location)
    }

    override fun hide(player: Player) {
        blockLine.hide(player)
    }

    override fun teleport(player: Player) {
        blockLine.teleport(player)
    }

    override fun show(player: Player) {
        blockLine.show(player)
    }

    override fun update(player: Player) {
        blockLine.update(player)
    }

    override fun setAnimation(animation: Animation, seeingPlayers: Collection<Player>) {
        animatedLine.setAnimation(animation, seeingPlayers)
    }

    override fun removeAnimation() {
        animatedLine.removeAnimation()
    }

    override var animation: Animation = animatedLine.animation
}
